src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/build.py

```diff
@@ -0,0 +1,230 @@
+                container_name=containerName, blob_name=blobName)
+            metadata = props.metadata
+            available = props.properties.content_length
+            lastModified = props.properties.last_modified
+
+        except AzureHttpError as ae:
+            if ae.status_code != 404:
+                sys.exit(ae)
+        except KeyboardInterrupt:
+            if len(currBytes) > 0:
+                print(currBytes.decode('utf-8'))
+            sys.exit()
+        except:
+            raise
+
+        # If we're still expecting data and we have a record for the last
+        # modified date and the last modified date has timed out, exit
+        if ((lastModified is not None and blob_is_not_complete(metadata)) or
+                start < available):
+
+            delta = datetime.utcnow().replace(tzinfo=pytz.utc) - lastModified
+
+            if delta.seconds > timeoutInSeconds:
+                print("No additional logs found. Timing out...")
+                sys.exit()
+
+        # If no new data available but not complete, sleep before trying
+        # to process additional data.
+        if (blob_is_not_complete(metadata) and start >= available):
+            time.sleep(5)
+
+
+def blob_is_not_complete(metadata):
+    if metadata is None:
+        return True
+
+    for key in metadata:
+        if key.lower() == 'complete':
+            return False
+
+    return True
+
+
+def acr_queue(cmd, client, registry_name, resource_group_name, source_location, docker_file_path, image_name, context_path=None, timeout=None, arguments=None, secret_arguments=None):
+    build_parameters = DockerBuildParameters(docker_file_path)
+
+    if not (context_path is None):
+        build_parameters.context_path = context_path
+
+    # [Server Bug] we should allow skiping image_name if users don't want to push
+    is_push_enabled = True
+    if image_name is None:
+        is_push_enabled = False
+
+    # hard-code platform to linux and cpu to 1
+    platform = PlatformProperties("Linux")
+    platform.cpu = 1
+
+    # figure out how to pass build_arguments
+    build_arguments = []
+    if not (arguments is None):
+        for name_value in arguments:
+            name, value = name_value.split('=', 1)
+            build_arguments.append(BuildArgument(name, value, False))
+
+    if not (secret_arguments is None):
+        for name_value in secret_arguments:
+            name, value = name_value.split('=', 1)
+            build_arguments.append(BuildArgument(name, value, True))
+
+    build_request = QueueBuildRequest(
+        image_name=image_name,
+        source_location=source_location,
+        build_parameters=build_parameters,
+        is_push_enabled=is_push_enabled,
+        timeout=timeout,
+        platform=platform,
+        build_arguments=build_arguments)
+
+    result = LongRunningOperation(cmd.cli_ctx)(client.queue(
+        build_request=build_request, resource_group_name=resource_group_name, registry_name=registry_name))
+
```

Comment:
Just return `acr_show_logs(cmd, client, registry_name, result.build_id, resource_group_name)` is fine, assuming acr_show_logs has proper error handling. In case of queue-build errors, `LongRunningOperation` would error out.

Reply:
But I wonder if you want to add a flag for users to decide if they want to see logs or just return so they don't have to wait.

Reply:
Yeah, we should add a flag for showing logs later. No reason to try/catch the exception correct?